eng/build.yml

```diff
@@ -179,19 +179,68 @@ jobs:
 
     - template: /eng/common/templates-official/variables/pool-providers.yml
 
-    steps:
-    - checkout: self
-      clean: true
-    - template: /eng/common/templates/steps/enable-internal-runtimes.yml
-    - ${{ if eq(parameters.agentOs, 'Windows_NT') }}:
-      - ${{ if and(not(parameters.isBuiltFromVmr), ne(variables['System.TeamProject'], 'public')) }}:
+    ${{ if and(not(parameters.isBuiltFromVmr), ne(variables['System.TeamProject'], 'public')) }}:
+      presteps:
+      # Install Azure CLI for enable-internal-runtimes.yml when the container is ubuntu2204 (the image is missing az).
+      - ${{ if and(eq(parameters.container, 'ubuntu2204'), eq(parameters.buildArchitecture, 'x64')) }}:
```

Comment:
Do you know if ubuntu 24.04 has the proper azure cli dep? If not, would it be better to see if 'ubuntu' is in the string of the container?

Reply:
The other containers we currently use seem to have the proper requirements to run the build (based on running test builds of this). Yesterday, I tried using a different container per recommendation, but it had other issues (didn't have Azure CLI). Tried installing that but that had issues, so I just reverted to this method of handling this container as further investment in this legacy repo build isn't worth it.